crates/graphql_network_protocol/src/graphql_network_protocol.rs

```diff
@@ -234,6 +242,24 @@ impl NetworkProtocol for GraphQLNetworkProtocol {
             {indent}}}"
         ))
     }
+
+    fn get_id_field_name(
+        db: &IsographDatabase<GraphQLNetworkProtocol>,
+        server_object_entity_name: ServerObjectEntityName,
+    ) -> DiagnosticResult<ServerScalarSelectableName> {
+        let entity = server_object_entity_named(db, server_object_entity_name)
+            .clone_err()?
+            .as_ref()
+            .ok_or_else(|| {
```

Comment:
let's re-use one of the existing diagnostics, `entity_not_defined_diagnostic`... actually I'll land a change to have a method `server_object_entity_named_result`, which returns a `Result` containing exactly that diagnostic.

Reply:
Ok, lemme know when you land it